src/stream.rs

```diff
@@ -452,8 +455,9 @@ impl OutputStream {
                 error_callback,
                 None,
             ),
-            _ => Err(cpal::BuildStreamError::StreamConfigNotSupported),
+            _ => return Err(StreamError::UnknownSampleFormat),
```

Comment:
This can only happen with incompatible CPAL version. I wonder why they decided to mark cpal sample format enum as `non_exhaustive`. This could probably panic instead.

Reply:
yeah had the same thought. But decided they must have done it for a reason. One of their examples (beep) has some commented out sample formats, so they might add those in the future (24bit for example)

Reply:
The reason I do not understand this is that cargo compiles both `rodio` and `cpal` from sources every time, so it can check that all match branches are there. Maybe there are some scenarios where pre-compiled Rust libraries are used, but I have not seen such projects so far.

Reply:
I recommend reading the RFC for `non_exhaustive`, it goes through all the reasons for using it. https://rust-lang.github.io/rfcs/2008-non-exhaustive.html

Reply:
and cpal is adding 24bit support to this enum it seems. Its non breaking because they have non_exhaustive on the enum. https://github.com/RustAudio/cpal/pull/927